packages/eslint-plugin/src/rules/strict-boolean-expressions.ts

```diff
@@ -814,6 +853,17 @@ export default util.createRule<Options, MessageId>({
         }
       }
 
+      const enums = types.filter(type =>
+        tsutils.isTypeFlagSet(
+          type,
+          ts.TypeFlags.Enum | ts.TypeFlags.EnumLike | ts.TypeFlags.EnumLiteral,
+        ),
+      );
+
+      if (enums.length) {
```

Comment:
`enums` is only used once, to check its `.length`. That means we can bail out of the loop early as soon as a match is found. A slight performance improvement:

```suggestion
      if (enums.some(enum => ...)) {
```

Reply:
Thank you for suggestion!
Actually I didn't know the performance difference between `.length` and `.some`.
Replaced!
https://github.com/typescript-eslint/typescript-eslint/pull/6096#:~:text=perf%3A%20use%20some%20instead%20of%20length